### DISCUSSION TITLE
How to access uploaded files

### DISCUSSION THREAD

**Discussion by HerzMichi** [zauberzeug/nicegui#1355] [Q&A] [answered]
### Question

```python
from nicegui import ui

ui.upload(on_upload=lambda e: ui.notify(f'Uploaded {e.name}')).classes('max-w-full')

ui.run()
``` 
This will upload a file and do a notification. 
But how can I access the file for storing it in a different folder. 
Thank you in advance :)

**Comment by falkoschindler** [comment]
Here is a demo how to show the uploaded `text`: https://nicegui.io/documentation/upload#show_file_content
I think you can start from there and write the text to a file. In case of binary data you probably don't need to `decode()` the file content.

**Reply by HerzMichi** [reply]
I want to upload a .csv-File and move it to a specified folder. 
Where is the uploaded file located. Is it in the [static files](https://nicegui.io/documentation#static_files) or somewhere else. 
I can't seem to find the location,

**Reply by HerzMichi** [reply, marked as answer]
This kind of worked. 
The upload conted is moved into a DataFrama and saves as a csv-File 
[Reddit](https://www.reddit.com/r/nicegui/comments/1590m7a/converting_an_uploaded_csv_file_to_a_pandas/)
